src/core/server/elasticsearch/status.ts

```diff
@@ -32,15 +32,26 @@ export const calculateStatus$ = (
           message,
           incompatibleNodes,
           warningNodes,
+          nodesInfoRequestError,
         }): ServiceStatus<ElasticsearchStatusMeta> => {
           if (!isCompatible) {
+            if (!nodesInfoRequestError) {
+              return {
+                level: ServiceStatusLevels.critical,
+                summary:
+                  // Message should always be present, but this is a safe fallback
+                  message ??
+                  `Some Elasticsearch nodes are not compatible with this version of Kibana`,
+                meta: { warningNodes, incompatibleNodes },
+              };
+            }
             return {
               level: ServiceStatusLevels.critical,
               summary:
                 // Message should always be present, but this is a safe fallback
                 message ??
                 `Some Elasticsearch nodes are not compatible with this version of Kibana`,
-              meta: { warningNodes, incompatibleNodes },
+              meta: { warningNodes, incompatibleNodes, nodesInfoRequestError },
```

Comment:
I wonder if we can write this in a different way so we avoid the duplicated code from lines 38-47.
```suggestion
              meta: { 
                warningNodes,
                incompatibleNodes,
                ...(nodesInfoRequestError && { nodesInfoRequestError }),
              },
```